internal/cli/command/cluster/feature/features/vault/common.go

```diff
@@ -0,0 +1,82 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vault
+
+import (
+	"emperror.dev/errors"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	featureName = "vault"
+
+	vaultCustom = "Custom vault"
+	vaultCP     = "Pipeline's Vault"
+
+	vaultSecretType = "vault"
+)
+
+type obj = map[string]interface{}
+
+type defaults struct {
+	address         string
+	secretID        string
+	policy          string
+	namespaces      []string
+	serviceAccounts []string
+}
+
+func NewDeactivateManager() *ActivateManager {
```

Comment:
I'd go with your idea to make a common base implementing the `GetName` method instead of returning an unrelated type. Or better yet, create a generic `DeactivateManager` that takes the feature name as parameter and implements the `GetName` method using it.